src/index.js

```diff
@@ -1,56 +1,101 @@
-  const code = `
-  <div class="list-item">
-    <div>
-      <input type="checkbox" id="item-${element.index}">
-      <p>${element.description}</p>
-    </div>
-      <button class="reset-btn">
-      <svg 
-      xmlns="http://www.w3.org/2000/svg" 
-      fill="none" 
-      viewBox="0 0 24 24" 
-      stroke-width="1.5" 
-      stroke="currentColor" 
-      class="delete-icon">
-    <path stroke-linecap="round" stroke-linejoin="round" 
-    d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0" />
-      </svg>
-      </button>
-  </div>
-  `;
-  listItem += code;
+let todo = JSON.parse(localStorage.getItem('todo')) || [];
+
+const renderTodoList = () => {
+  let listItem = '';
+  todo.forEach((element, index) => {
+    const code = `
+      <div class="list-item">
+        <div>
+          <input type="checkbox" class="check" id="item-${index}">
+          <input type="text" class="edit-input" id="input-${index}" value="${element.description}">
+        </div>
+        <button class="reset-btn delete-btn" id="${index}">
+          <svg 
+            xmlns="http://www.w3.org/2000/svg" 
+            fill="none" 
+            viewBox="0 0 24 24" 
+            stroke-width="1.5" 
+            stroke="currentColor" 
+            class="delete-icon">
+            <path stroke-linecap="round" 
+            stroke-linejoin="round" 
+              d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0" />
+          </svg>
+        </button>
+      </div>
+    `;
+    listItem += code;
+  });
+  listEl.innerHTML = listItem;
+
+  const deleteButtons = document.querySelectorAll('.delete-btn');
+  deleteButtons.forEach((button) => {
+    button.addEventListener('click', () => {
+      const index = parseInt(button.id, 20);
+      // eslint-disable-next-line no-use-before-define
+      deleteItem(index);
+    });
+  });
+
+  const editInputs = document.querySelectorAll('.edit-input');
+  editInputs.forEach((input) => {
+    input.addEventListener('change', () => {
+      const index = parseInt(input.id.split('-')[1], 10);
+      const newDescription = input.value;
+      // eslint-disable-next-line no-use-before-define
+      updateDescription(index, newDescription);
+    });
+  });
+};
+
+// DELETE ITEM FROM LIST
+const deleteItem = (index) => {
```

Comment:
- [optional] I can see you created a module folder. I will advise you remove some methods (add,remove and edit) into it and import them instead to your index file. This will make your work more organized.